pgcli/pgexecute.py

```diff
@@ -215,6 +221,9 @@ def connect(self, database=None, user=None, password=None, host=None,
         self.host = host
         self.port = port
 
+        if not self.host:
+            self.host = self.socket_directory()
```

Comment:
I assumed that since it was postgres behavior, it would be an issue: https://www.postgresql.org/docs/9.6/static/libpq-connect.html#LIBPQ-CONNECT-HOST

But if you feel more confortable, we can separate both. It'll require some work since when you use `-h /socket_dir` you'll have a socket directory in the `host` property already.

My changes are there mostly to handle default behavior, when no host is passed to the connect method.